csharp/src/Apache.Arrow/C/CArrowSchemaImporter.cs

```diff
@@ -0,0 +1,285 @@
+
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Apache.Arrow.Types;
+
+namespace Apache.Arrow.C
+{
+    public static class CArrowSchemaImporter
+    {
+        /// <summary>
+        /// Import C pointer as an <see cref="ArrowType"/>.
+        /// </summary>
+        /// <examples>
+        /// Typically, you will allocate a uninitialized CArrowSchema pointer,
+        /// pass that to external function, and then use this method to import
+        /// the result.
+        /// 
+        /// <code>
+        /// CArrowSchema* importedPtr = CArrowSchema.New();
+        /// foreign_export_function(importedPtr);
+        /// ArrowType importedType = CArrowSchemaImporter.ImportType(importedPtr);
+        /// CArrowSchema.Free(importedPtr);
+        /// </code>
+        /// </examples>
+        public static unsafe ArrowType ImportType(CArrowSchema* ptr)
+        {
+            using var importedType = new ImportedArrowSchema(ptr);
+            return importedType.GetAsType();
+        }
+
+        /// <summary>
+        /// Import C pointer as an <see cref="Field"/>.
+        /// </summary>
+        /// <examples>
+        /// Typically, you will allocate a uninitialized CArrowSchema pointer,
+        /// pass that to external function, and then use this method to import
+        /// the result.
+        /// 
+        /// <code>
+        /// CArrowSchema* importedPtr = CArrowSchema.New();
+        /// foreign_export_function(importedPtr);
+        /// Field importedField = CArrowSchemaImporter.ImportField(importedPtr);
+        /// CArrowSchema.Free(importedPtr);
+        /// </code>
+        /// </examples>
+        public static unsafe Field ImportField(CArrowSchema* ptr)
+        {
+            using var importedField = new ImportedArrowSchema(ptr);
+            return importedField.GetAsField();
+        }
+
+        /// <summary>
+        /// Import C pointer as an <see cref="Schema"/>.
+        /// </summary>
+        /// <examples>
+        /// Typically, you will allocate a uninitialized CArrowSchema pointer,
+        /// pass that to external function, and then use this method to import
+        /// the result.
+        /// 
+        /// <code>
+        /// CArrowSchema* importedPtr = CArrowSchema.New();
+        /// foreign_export_function(importedPtr);
+        /// Field importedSchema = CArrowSchemaImporter.ImportSchema(importedPtr);
+        /// CArrowSchema.Free(importedPtr);
+        /// </code>
+        /// </examples>
+        public static unsafe Schema ImportSchema(CArrowSchema* ptr)
+        {
+            using var importedSchema = new ImportedArrowSchema(ptr);
+            return importedSchema.GetAsSchema();
+        }
+
+        private sealed unsafe class ImportedArrowSchema : IDisposable
+        {
+            private readonly CArrowSchema* _data;
+            private readonly bool _isRoot;
+
+            public ImportedArrowSchema(CArrowSchema* handle)
```

Comment:
Are we checking the `handle` pointer for `null`? This is passed directly here from a public API.